### ISSUE TITLE
persistent bug in find_proposition()

### ISSUE THREAD

**Issue by benbrastmckie** [benbrastmckie/ModelChecker#51]
I am getting the following error report:
```
SENT OP \neg
SENT ARGS [(A \vee B)]
SENT ARGS TYPES <class 'syntactic.Sentence'>
Traceback (most recent call last):
  File "/home/benjamin/Documents/Philosophy/Projects/ModelChecker/Code/src/new_ch
ecker/test_example.py", line 174, in <module>
    model_structure = ModelStructure(
                      ^^^^^^^^^^^^^^^
  File "/home/benjamin/Documents/Philosophy/Projects/ModelChecker/Code/src/new_ch
ecker/hidden_things.py", line 327, in __init__
    self.interpret(self.syntax.all_sentences.values())
  File "/home/benjamin/Documents/Philosophy/Projects/ModelChecker/Code/src/new_ch
ecker/hidden_things.py", line 373, in interpret
    sent_obj.update_proposition(self)
  File "/home/benjamin/Documents/Philosophy/Projects/ModelChecker/Code/src/new_ch
ecker/syntactic.py", line 89, in update_proposition
    self.proposition = model_structure.proposition_class(self, model_structure)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/benjamin/Documents/Philosophy/Projects/ModelChecker/Code/src/new_ch
ecker/exposed_things.py", line 199, in __init__
    self.verifiers, self.falsifiers = self.find_proposition()
                                      ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/benjamin/Documents/Philosophy/Projects/ModelChecker/Code/src/new_ch
ecker/exposed_things.py", line 342, in find_proposition
    return self.operator.find_verifiers_and_falsifiers(*self.arguments)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: NegationOperator.find_verifiers_and_falsifiers() missing 1 required po
sitional argument: 'leftarg'
```
Here is the function which includes line 341 from `exposed_things.py`:
```
def find_proposition(self):
    all_bits = self.model_structure.all_bits
    model = self.model_structure.z3_model
    sem = self.semantics
    if len(self.prefix_string) == 1:
        if self.prefix_string[0] in {'\\top', '\\bot'}:
            operator = self.prefix_sentence[0]
            return operator.find_verifiers_and_falsifiers()
        if self.prefix_string[0].isalnum():
            print("CHECK", self.prefix_string)
            sentence_letter = self.prefix_sentence[0]
            V = {
                bit for bit in all_bits
                if model.evaluate(sem.verify(bit, sentence_letter))
            }
            F = {
                bit for bit in all_bits
                if model.evaluate(sem.falsify(bit, sentence_letter))
            }
            return V, F
        raise ValueError(
            f"Their is no proposition for {self.prefix_string[0]}."
        )
    print("SENT OP", self.operator.name)
    print("SENT ARGS", self.arguments)
    print("SENT ARGS TYPES", type(self.arguments[0]))
    return self.operator.find_verifiers_and_falsifiers(*self.arguments)
```
It is clearly looking up the right operator (negation) but struggling to find any arguments. However, the argument also looks right: it's a sentence object for `(A \vee B)`. But somehow it is unable to pass this argument into the class method. I have tried leaving off the star, and various other things, but no luck yet.

**Comment by benbrastmckie** [comment]
Started to fix the bug. It was because the operator was not instantiated (it was a class not an instance). I fixed that, but am now facing a trickier issue which is that to find the proposition for a complex sentence, it looks for what the proposition is for it's arguments, but these haven't been updated upon looking for the proposition for the first sentence.

One idea was to try to instantiate all sentences by updating the proposition for each starting with lowest complexity and moving upwards. I can see how this might go, but wondering if there is a better solution where the updating happens somewhat more naturally (recursively) without having to use brute force.

I added HINTS to search for where the relevant parts are in `find_verifiers_and_falsifiers()` in the operators, e.g., `NegationOperator`, as well as `find_propostion()` in `Proposition`.
